core/models.py

```diff
@@ -439,15 +467,51 @@ def save(self, *args, **kwargs):
         if self.end_date and timezone.is_naive(self.end_date):
             self.end_date = timezone.make_aware(self.end_date)
 
+        if self.review_status == ReviewStatus.Approved:
+            self.is_public = True
+        if self.review_status == ReviewStatus.Rejected:
+            self.is_public = False
+
+        h = Hackathon.objects.filter(
+            name=self.name,
+            end_date__year=self.end_date.year,
+            end_date__month=self.end_date.month,
+        )
+        if len(h) > 0 and h[0].id != self.id:
+            raise ValidationError(
+                f"An object with the name '{self.name}' already exists."
+            )
```

Comment:
why not just so dup id ? its prob faster than query the db
```suggestion
        h = Hackathon.objects.filter(
            name=self.name,
            end_date__year=self.end_date.year,
            end_date__month=self.end_date.month,
        )
        if len(h) > 0 and h[0].id != self.id:
            raise ValidationError(
                f"An object with the name '{self.name}' already exists."
            )
```

Reply:
James... it's your code

Reply:
that's amol's code

Reply:
Then don't comment about it on your PR. amol has one open